server/tests/chat/test_after_rca_dispatch.py

```
@@ -0,0 +1,109 @@
+
+class TestAfterRcaDispatchDeduplication:
+    """Validate the flag-based contract that prevents double-dispatch."""
+
+    def _simulate_outer_dispatch_decision(self, result: dict, incident_id: str, trigger_metadata: dict) -> bool:
+        """Replicate the outer function's dispatch decision logic.
+        
+        Returns True if the outer function WOULD dispatch after_rca actions.
+        This mirrors the exact condition in run_background_chat().
+        """
+        if incident_id and trigger_metadata and trigger_metadata.get('source') != 'action':
+            if not result.get('after_rca_dispatched'):
```

Ruff (0.15.15)

[warning] Use a single `if` statement instead of nested `if` statements

(SIM102)